engine/table/src/main/java/io/deephaven/engine/table/impl/by/PartitionByChunkedOperator.java

```diff
@@ -90,13 +91,17 @@ public interface AttributeCopier {
     private final String callSite;
 
     private final ObjectArraySource<QueryTable> tables;
-    private final ObjectArraySource<WritableRowSet> addedRowSets;
-    private final ObjectArraySource<WritableRowSet> removedRowSets;
-    private final ObjectArraySource<WritableRowSet> modifiedRowSets;
+    private final ObjectArraySource<RowSetBuilderSequential> addedBuildersSequential;
+    private final ObjectArraySource<RowSetBuilderRandom> addedBuilders;
+    private final ObjectArraySource<RowSetBuilderRandom> removedBuilders;
+    private final ObjectArraySource<RowSetBuilderRandom> modifiedBuilders;
+
     private final ObjectArraySource<RowSetShiftData.SmartCoalescingBuilder> shiftDataBuilders;
     private final ModifiedColumnSet resultModifiedColumnSet;
     private final ModifiedColumnSet.Transformer upstreamToResultTransformer;
 
+    private volatile boolean initialized;
```

Comment:
Need not be `volatile`; arguably same applies to the other two `volatile` fields. I'm fairly convinced of the happens-before relationship between initialization and subsequent update processing without the additional protection.